### PR TITLE
Fix check for database corruption.

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Sometimes when the node crashes or is killed the blockstate.dat ends up of the correct size, but with trailing zeros.
This PR fixes the check for "database is good" by attempting to load the block state at the given location.

## Changes

- Make the check for "is database corrupted" more strict.
- Bump version number for next release.

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs [hunk 1]
```diff
@@ -347,8 +347,8 @@ readBlobBS bs@BlobStoreAccess{..} br@(BlobRef offset) = do
 
 -- |Check if a given 'BlobRef' is valid in the blob store. This does not attempt to deserialize
```
Comment (mkmks) [inline concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs:348]: I guess, this comment is not accurate after having changed `readBlobBS` to `loadRef`. The former does not deserialize the blob, but the latter does.

**Review by mkmks — APPROVED** [review]
Just out of curiosity, why would a check based on `readBlobBS` fail where a check based on `loadRef` succeed? The difference between the two is deserialization. Are we expecting that it would not be possible to deserialize non-truncated blobs?

**Review by EmilLa1 — APPROVED** [review]

File: concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs (hunk 1 above)
Comment (abizjak) [reply concordium-consensus/src/Concordium/GlobalState/Persistent/BlobStore.hs]: I fixed the comment.

**Comment by abizjak** [comment]
> Just out of curiosity, why would a check based on `readBlobBS` fail where a check based on `loadRef` succeed? The difference between the two is deserialization. Are we expecting that it would not be possible to deserialize non-truncated blobs?

I tried to explain in the PR description. The blobstore ends with enough trailing zeros.

**Pull request merged**